src/filesystem/interface.js

```diff
@@ -206,6 +206,60 @@ function FileSystem(options, callback) {
     return watcher;
   };
 
+  //Object that uses filenames as keys
+  const statWatchers = new Map();
+
+  this.watchFile = function (filename, options, listener) {
+    let prevStat, currStat;
+
+    if (Path.isNull(filename)) {
+      throw new Error('Path must be a string without null bytes.');
+    }
+    // Checks to see if there were options passed in and if not, the callback function will be set here
+    if (typeof options === 'function') {
+      listener = options;
+      options = {};
+    }
+    // default 5007ms interval, persistent is not used this project
+    const interval = options.interval || 5007;
+    listener = listener || nop;
+
+    let intervalValue = statWatchers.get(filename);
+
+    if(intervalValue) {
+      return;
+    }
+    else { 
+      fs.stat(filename, function (err, stats) {
```

Comment:
There's some kind of timing bug here.  You do a `stat`, and then inside the callback you start your interval, but never seem to call `stat` again in that interval's callback.  Each interval should recall stat, so you have updated values for prev/currStat.

Also, this raises the question about the `interval`'s lower bounds.  There is some amount of time that a `stat` will take to execute.  We shouldn't trigger another one before it's done.  Probably we should ignore requests for `interval` to be lower than `5007` and just use that as our floor.